<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Workshop-popgenome by tonig-evo</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Workshop-popgenome</h1>
      <h2 class="project-tagline">Tutorial about R package PopGenome</h2>
      <a href="https://github.com/tonig-evo/workshop-popgenome" class="btn">View on GitHub</a>
      <a href="https://github.com/tonig-evo/workshop-popgenome/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/tonig-evo/workshop-popgenome/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <p><a href="http://tonig-evo.github.io/workshop-popgenome">http://tonig-evo.github.io/workshop-popgenome</a></p>

<h1>
<a id="introduction" class="anchor" href="#introduction" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Introduction</h1>

<p>PopGenome is an R package
(<a href="https://cran.r-project.org/web/packages/PopGenome/index.html">https://cran.r-project.org/web/packages/PopGenome/index.html</a>) for
analyses of population genomic data. For this
tutorial, please make sure that your R working directory is set
correctly and you have all the packages installed, e.g.
install.packages("PopGenome"). The following files are necessary to
conduct this practical session:</p>

<ul>
<li><p>fasta_file.txt, a fasta file for one locus from different
<em>Arabidopsis thaliana</em> individuals (accessions) and the outgroup
sequence from <em>Arabidopsis lyrata</em> in the folder <strong>fasta</strong></p></li>
<li><p>LGE22.gff, LGE22.vcf, LGE22.fa in subfolders in the folder
<strong>great_tit</strong></p></li>
<li><p>variants.vcf, ind_species1.txt, ind_species2.txt, rad_assembly.fa
in the folder <strong>rad</strong></p></li>
</ul>

<p>To get an overview about the file contents, inspect files with a text
Editor (e.g. Notepad+) or via the command line (e.g. more) before
starting to work on them.</p>

<h1>
<a id="load-library-into-r" class="anchor" href="#load-library-into-r" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Load library into R</h1>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Loading module </span>
library(<span class="pl-smi">PopGenome</span>)</pre></div>

<h1>
<a id="fasta-files" class="anchor" href="#fasta-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Fasta Files</h1>

<h2>
<a id="reading-fasta-files" class="anchor" href="#reading-fasta-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Reading Fasta Files</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Reading data, read fasta from folder </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> readData(<span class="pl-s"><span class="pl-pds">"</span>fasta<span class="pl-pds">"</span></span>) 
get.sum.data(<span class="pl-smi">GENOME.class</span>)
get.individuals(<span class="pl-smi">GENOME.class</span>)</pre></div>

<p>1.What statistics can one obtain from get.sum.data function?</p>

<p>2.Folders <strong>fasta_a</strong>, <strong>fasta_b</strong>, <strong>fasta_c</strong> contain modified alignments. Identify the differences between the datasets. Why PopGenome fails to load fasta files?</p>

<h2>
<a id="obtaining-summary-statistics-from-alignments" class="anchor" href="#obtaining-summary-statistics-from-alignments" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Obtaining summary statistics from alignments</h2>

<p><strong>Note:</strong> Since calculation of certain population genetic parameter can
be computational intense, they have to be executed separately
beforehand. For this modules have to be run. Note that module <strong>Fst</strong>
has to be executed with <strong>F_st</strong>. The statistic Tajimaâ€™s D is part of
the module <strong>neutrality</strong> not <strong>Fst</strong></p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Available statistics and examples </span>
show.slots(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-c"># Run necessary module </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> F_ST.stats(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> neutrality.stats(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">n.sites</span> 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">Pi</span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">Tajima.D</span></pre></div>

<p>3.What different modules are available? (show.slots)</p>

<p>4.What module is necessary to be executed in order to obtain Wall.B?</p>

<p>5.How could one obtain a per site estimate of Pi?</p>

<h2>
<a id="obtaining-statistics-for-regions" class="anchor" href="#obtaining-statistics-for-regions" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Obtaining statistics for regions</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Available region data and statistics </span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.stats</span> 
<span class="pl-c"># Examples</span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">biallelic.sites</span>[[<span class="pl-c1">1</span>]][<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>]
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">transitions</span>[[<span class="pl-c1">1</span>]][<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">10</span>]</pre></div>

<p>6.How many sites have gaps?</p>

<p>7.How many singletons are in the dataset? (see also <strong>An_introduction_to_the_PopGenome_package.pdf</strong>, section
3.1)</p>

<p>8.What is the difference between <em>region.data</em> and <em>region.stats</em>? (see also <strong>Whole_genome_analyses_using_VCF_files.pdf</strong>, section 11 and 12)</p>

<h2>
<a id="define-outgroups-and-populations" class="anchor" href="#define-outgroups-and-populations" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Define outgroups and populations</h2>

<p><strong>Note:</strong> If one ore more outgroup sequences are defined, PopGenome will
only consider SNPs where the outgroup is monomorphic; the monomorphic
nucleotide is then automatically defined as the major allele (encoded by
0).</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Without defining populations </span>
get.individuals(<span class="pl-smi">GENOME.class</span>)
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> neutrality.stats(<span class="pl-smi">GENOME.class</span>,<span class="pl-v">detail</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
get.neutrality(<span class="pl-smi">GENOME.class</span>)[[<span class="pl-c1">1</span>]] 
<span class="pl-c"># Define populations with lists</span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> set.populations(<span class="pl-smi">GENOME.class</span>,<span class="pl-k">list</span>(
c(<span class="pl-s"><span class="pl-pds">"</span>CON<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>KAS-1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>RUB-1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>PER-1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>RI-0<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>MR-0<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>TUL-0<span class="pl-pds">"</span></span>),
c(<span class="pl-s"><span class="pl-pds">"</span>MH-0<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>YO-0<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>ITA-0<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>CVI-0<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>COL-2<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>LA-0<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>NC-1<span class="pl-pds">"</span></span>) )) 
<span class="pl-c"># Check whether grouping is set correctly </span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">populations</span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">populations2</span> 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">outgroup</span>
<span class="pl-c"># Recalculate statistics for populations </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span>neutrality.stats(<span class="pl-smi">GENOME.class</span>,<span class="pl-v">detail</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">Tajima.D</span> 
<span class="pl-c"># Each population </span>
get.neutrality(<span class="pl-smi">GENOME.class</span>)[[<span class="pl-c1">1</span>]]
get.neutrality(<span class="pl-smi">GENOME.class</span>)[[<span class="pl-c1">2</span>]] 
<span class="pl-c"># Set an outgroup </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span>set.outgroup(<span class="pl-smi">GENOME.class</span>,c(<span class="pl-s"><span class="pl-pds">"</span>Alyrata<span class="pl-pds">"</span></span>))
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">outgroup</span> <span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> neutrality.stats(<span class="pl-smi">GENOME.class</span>,<span class="pl-v">detail</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
get.neutrality(<span class="pl-smi">GENOME.class</span>)[[<span class="pl-c1">1</span>]] 
get.neutrality(<span class="pl-smi">GENOME.class</span>)[[<span class="pl-c1">2</span>]</pre></div>

<p>9.Name implemented statistics that require an outgroup, e.g. that are
calculated after defining the outgroup.</p>

<p>10.What do you have to pay attention to when applying the McDonald-Kreitman
test?\
(see <strong>Whole_genome_analyses_using_VCF_files.pdf</strong>)</p>

<h1>
<a id="analysing-vcf-files-for-whole-genome-data" class="anchor" href="#analysing-vcf-files-for-whole-genome-data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Analysing VCF files for whole genome data</h1>

<h2>
<a id="loading-vcf-files" class="anchor" href="#loading-vcf-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Loading VCF files</h2>

<p>There are two ways to read in VCF files, either a folder of VCF files
with <strong>readData</strong> or a single VCF file with <strong>readVCF</strong>. To read a VCF
file using <em>readVCF</em> it needs to be compressed with <em>bgzip</em> and indexed
with <em>tabix</em>. The tabix files need to be placed in the same folder as
the vcf file.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># What parameters need to be defined </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span>readVCF(<span class="pl-s"><span class="pl-pds">"</span>great_tit/vcf/LGE22.vcf.gz<span class="pl-pds">"</span></span>, <span class="pl-c1">6000</span>,<span class="pl-s"><span class="pl-pds">"</span>chrLGE22_Parus_Major_build_1.0.2<span class="pl-pds">"</span></span>,<span class="pl-c1">1</span>,<span class="pl-c1">773534</span>)
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.names</span> 
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> neutrality.stats(<span class="pl-smi">GENOME.class</span>, <span class="pl-v">FAST</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) 
get.sum.data(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span></pre></div>

<p>11.What parameters need to be defined to <em>readVCF</em>? (see <strong>PopGenome.pdf</strong>)</p>

<p>12.What is the overall diversity (theta and pi) of chromosome LGE22?</p>

<h2>
<a id="loading-vcf-files-with-annotation" class="anchor" href="#loading-vcf-files-with-annotation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Loading VCF files with annotation</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-smi">GENOME2.class</span> <span class="pl-k">&lt;-</span> readData(<span class="pl-s"><span class="pl-pds">"</span>great_tit/vcf2<span class="pl-pds">"</span></span>,<span class="pl-v">format</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>VCF<span class="pl-pds">"</span></span>, <span class="pl-v">gffpath</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>great_tit/gff<span class="pl-pds">"</span></span>) 
get.sum.data(<span class="pl-smi">GENOME2.class</span>)
<span class="pl-smi">GENOME2.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span> 
<span class="pl-smi">GENOME2.class</span> <span class="pl-k">&lt;-</span> set.synnonsyn(<span class="pl-smi">GENOME2.class</span>, <span class="pl-v">ref.chr</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>great_tit/fasta/LGE22.fasta<span class="pl-pds">"</span></span>)
<span class="pl-smi">GENOME2.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">synonymous</span>
<span class="pl-smi">GENOME2.class</span><span class="pl-k">@</span><span class="pl-smi">region.data</span><span class="pl-k">@</span><span class="pl-smi">CodingSNPS</span> 
<span class="pl-smi">GENOME2.class.syn</span> <span class="pl-k">&lt;-</span> neutrality.stats(<span class="pl-smi">GENOME2.class</span>,<span class="pl-v">subsites</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>syn<span class="pl-pds">"</span></span>)
<span class="pl-smi">GENOME2.class.syn</span><span class="pl-k">@</span><span class="pl-smi">Tajima.D</span> 
<span class="pl-smi">GENOME2.class.syn</span><span class="pl-k">@</span><span class="pl-smi">theta_Watterson</span></pre></div>

<p>13.What is theta Watterson and Tajimaâ€™s D of chromosome LGE22 for synonymous and nonsynonymous sites?</p>

<h1>
<a id="analysing-radseq-data-using-vcf-" class="anchor" href="#analysing-radseq-data-using-vcf-" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Analysing RADseq data using VCF </h1>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># SPLIT VCF FILE</span>
VCF_split_into_scaffolds(<span class="pl-s"><span class="pl-pds">"</span>rad/variants.vcf<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>rad_split_vcf<span class="pl-pds">"</span></span>) 
<span class="pl-c"># READ IN DATA, smaller subset</span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> readData(<span class="pl-s"><span class="pl-pds">"</span>rad_split_vcf_small<span class="pl-pds">"</span></span>,<span class="pl-v">format</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>VCF<span class="pl-pds">"</span></span>) 
<span class="pl-smi">pop1</span><span class="pl-k">&lt;-</span>as.character(read.table(<span class="pl-s"><span class="pl-pds">"</span>rad/ind_species1.txt<span class="pl-pds">"</span></span>)[[<span class="pl-c1">1</span>]]) 
<span class="pl-smi">pop2</span><span class="pl-k">&lt;-</span>as.character(read.table(<span class="pl-s"><span class="pl-pds">"</span>rad/ind_species2.txt<span class="pl-pds">"</span></span>)[[<span class="pl-c1">1</span>]]) 
<span class="pl-smi">GENOME.class</span><span class="pl-k">&lt;-</span> set.populations(<span class="pl-smi">GENOME.class</span>,<span class="pl-k">list</span>(<span class="pl-smi">pop1</span>,<span class="pl-smi">pop2</span>),<span class="pl-v">diploid</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) 
<span class="pl-c"># CHECK</span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">populations</span></pre></div>

<h2>
<a id="obtaining-statistics-from-multiple-vcfs-derived-from-radseq" class="anchor" href="#obtaining-statistics-from-multiple-vcfs-derived-from-radseq" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Obtaining statistics from multiple VCFs derived from RADseq</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># NEUTRALITY STATISTICS </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> neutrality.stats(<span class="pl-smi">GENOME.class</span>, <span class="pl-v">FAST</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) 
get.neutrality(<span class="pl-smi">GENOME.class</span>)[[<span class="pl-c1">1</span>]] <span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">Tajima.D</span> 
<span class="pl-c"># FST </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> F_ST.stats(<span class="pl-smi">GENOME.class</span>)
get.F_ST(<span class="pl-smi">GENOME.class</span>)[[<span class="pl-c1">1</span>]] 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">nucleotide.F_ST</span> 
<span class="pl-c"># DIVERSITY</span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> diversity.stats(<span class="pl-smi">GENOME.class</span>)
get.diversity(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">nuc.diversity.within</span> 
<span class="pl-c"># SFS</span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span>detail.stats(<span class="pl-smi">GENOME.class</span>,<span class="pl-v">site.spectrum</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,<span class="pl-v">site.FST</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) 
<span class="pl-smi">results</span> <span class="pl-k">&lt;-</span> get.detail(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">region.stats</span><span class="pl-k">@</span><span class="pl-smi">minor.allele.freqs</span></pre></div>

<p>14.Plot a site-frequency-spectrum for each population.</p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Concatenate loci</span>
<span class="pl-smi">CON</span> <span class="pl-k">&lt;-</span> concatenate.regions(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-smi">CON</span> <span class="pl-k">&lt;-</span> detail.stats(<span class="pl-smi">CON</span>,<span class="pl-v">site.spectrum</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,<span class="pl-v">site.FST</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>) 
<span class="pl-smi">results</span> <span class="pl-k">&lt;-</span>get.detail(<span class="pl-smi">CON</span>) 
<span class="pl-smi">allele_Freqs</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">CON</span><span class="pl-k">@</span><span class="pl-smi">region.stats</span><span class="pl-k">@</span><span class="pl-smi">minor.allele.freqs</span>[[<span class="pl-c1">1</span>]] 
<span class="pl-smi">freq.table</span> <span class="pl-k">&lt;-</span> <span class="pl-k">list</span>()
<span class="pl-smi">freq.table</span>[[<span class="pl-c1">1</span>]] <span class="pl-k">&lt;-</span> table(<span class="pl-smi">allele_Freqs</span>) 
<span class="pl-smi">sfs</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-smi">freq.table</span>)

library(<span class="pl-smi">ggplot2</span>) 
ggplot(<span class="pl-smi">sfs</span>, aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">allele_Freqs</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">Freq</span>)) <span class="pl-k">+</span> geom_bar(<span class="pl-v">stat</span> <span class="pl-k">=</span> â€™<span class="pl-smi">identity</span>â€™)</pre></div>

<h1>
<a id="additional-aspects" class="anchor" href="#additional-aspects" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Additional aspects</h1>

<p>More information are available in three pdfs accompanied by the program
(see folder <strong>pdf</strong>): <strong>An introduction to the PopGenome package</strong>:
Sliding window analysis, reading SNP data files, coalescent simulations;
<strong>Whole genome analyses using PopGenome and VCF files</strong>: Details about
reading tabixed VCF files, examples, graphical output, parallel read-in,
pre-filtering VCF files; <strong>Package PopGenome</strong>: Documentation about all
implemented functions with examples</p>

<h2>
<a id="including-features-from-gff-files-to-fasta-files" class="anchor" href="#including-features-from-gff-files-to-fasta-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Including features from GFF files to Fasta files</h2>

<p>If no gff-file was specified when the data was read in, it is assumed
that the alignment is in the correct reading frame (starting at a first
codon position). The GFF folder contains GFF-files for each alignment
stored in the FASTA folder. The GFF files should have the same names
(without any extensions like .fas or .gff) as the corresponding FASTA
files to ensure that sequence and annotation are matched correctly.</p>

<h2>
<a id="handling-missing-data-and-differences-between-readdata-and-readvcf" class="anchor" href="#handling-missing-data-and-differences-between-readdata-and-readvcf" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Handling missing data and differences between readData and readVCF</h2>

<p>PopGenome can use missing data, e.g. positions with gaps. Also note the
differences between <em>readData</em> and <em>readVCF</em> for your own analysis (for
further details see <strong>Whole_genome_analyses_using_VCF_files.pdf</strong>).</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/tonig-evo/workshop-popgenome">Workshop-popgenome</a> is maintained by <a href="https://github.com/tonig-evo">tonig-evo</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
